src/bio.c

```diff
@@ -580,8 +580,8 @@ int wolfSSL_BIO_gets(WOLFSSL_BIO* bio, char* buf, int sz)
                 const byte* c;
                 int   cSz;
                 cSz = wolfSSL_BIO_pending(bio);
-                if (cSz < 0) {
-                    ret = cSz;
+                if (cSz <= 0) {
+                    ret = (cSz == 0) ? WOLFSSL_BIO_ERROR : cSz;
```

Comment:
Did you confirm the `BIO_gets` compatibility allows for returning `WOLFSSL_BIO_ERROR`? I have a feeling it is supposed to return 0 in this case. I'd also check with @JacobBarthelmeh and see if he knows.